# EOSC 582 Assignment II (Petty, p128, 154, 168) - Due Feb 13

In [1]:
__author__ = 'Yingkai (Kyle) Sha'
__email__  = 'yingkai@eos.ubc.ca' 

# Content
1. [**Problem 6.9**](#Problem-6.9)
1. [**Problem 6.28**](#Problem-6.28)
1. [**Problem 7.4**](#Problem-7.4)

In [2]:
import numpy as np
import matplotlib.pyplot as plt
from __future__ import print_function
% matplotlib inline

# Problem 6.9

**Q1: Calculate $\mathrm{T_B}$**

We can use inverse Planck function for $\mathrm{T_B}$:

\begin{equation*}
    T_B\left(\lambda, L_e\right) = \frac{2h/k_B}{\lambda \ln{\left( 2hc^2/\lambda^5 L_e+1 \right)}}
\end{equation*}

$h$: Planck constant; $k_B$: Boltzmann constant; $c$: Speed of light.

Given conditions:

In [3]:
wave_length = 12*1e-6 # m
radiance    = 6.2*1e6 # W/m^2/m/sr

Constants:

In [4]:
c=2.99792458e+08  # m/s, Speed of light
h=6.62606876e-34  # Js, Planck constant
kb=1.3806503e-23  # J/K, Boltzman constant

Calculate $\mathrm{T_B}$

In [5]:
TB = (h*c/kb)/(wave_length*np.log((2 * h * c**2)/(wave_length**5 * radiance) + 1))
print('Brightness Temperature: {}'.format(TB))

Brightness Temperature: 275.039598465


** Q2: Calculate real temperature by using emissivity** 

If the surface is in a thermal equilibrium state, we can use *Kirchhoff's law* and get the following relation:

\begin{equation*} 
    L_e = \epsilon \cdot L_b
\end{equation*}

$\mathrm{L_e}$: "true" radiance; $\mathrm{L_b}$: "equvilant" black body radiance.

Thus we get the right black body radiance for inverse Planck function.

In [6]:
emiss = 0.9
radiance = radiance / emiss

In [7]:
T_true = (h*c/kb)/(wave_length*np.log((2 * h * c**2)/(wave_length**5 * radiance) + 1))
print('True Temperature: {}'.format(TB))

True Temperature: 275.039598465


** Q3: Ratio of brightness temperature and true temperature **

In [8]:
TB/T_true

0.97615654535521668

The ratio is **not equal** to emissivity, and we can prove that by play with Planck function.

# Problem 6.28

In [9]:
Freq = np.array([    6.93    ,     10.65   ,    18.70    ,    23.80    ,    36.50    ,     89.00   ])
N_10 = np.array([8.095+2.371j, 7.431+2.708j, 6.164+2.980j, 5.566+2.958j, 4.575+2.721j, 3.115+1.842j])
N_20 = np.array([8.211+2.144j, 7.745+2.416j, 6.712+2.819j, 6.151+2.901j, 5.123+2.831j, 3.433+2.083j])

** Q1: Calculate brightness temperature for all cases**

Since the surface is smooth, we can use *Fresnel relations* for reflectivity:

* For vertical polarization:
\begin{equation*}
    R_p = \left| \frac{cos \Phi_t - Ncos \Phi_i}{cos \Phi_t + Ncos \Phi_i} \right|^2
\end{equation*}
* For horizontal polarization:
\begin{equation*}
    R_s = \left| \frac{cos \Phi_i - Ncos \Phi_t}{cos \Phi_i + Ncos \Phi_t} \right|^2
\end{equation*}

Here $\mathrm \Phi_i = 55 ^ \circ $. According to *Snell's Law*:
\begin{equation*}
    \Phi_t = arcsin\left( \frac{sin\left( \Phi_i \right)}{N} \right)
\end{equation*}

Then we can calculate $\mathrm{T_B}$ as:

* for horizontal channel:

\begin{equation*}
    T_{B} =  \epsilon_s T = \left( 1 - R_s \right) T
\end{equation*}

* for vertical channel:

\begin{equation*}
    T_{B} = \epsilon_p T = \left( 1 - R_p \right) T
\end{equation*}

In [10]:
def TBright_calc(theta_i, N, T):
    theta_t = np.arcsin( np.sin(theta_i)/N )
    Rp = np.abs((np.cos(theta_t) - N*np.cos(theta_i))/(np.cos(theta_t) + N*np.cos(theta_i)))**2
    Rs = np.abs((np.cos(theta_i) - N*np.cos(theta_t))/(np.cos(theta_i) + N*np.cos(theta_t)))**2
    TBH = T * (1-Rs)
    TBV = T * (1-Rp)
    return TBH, TBV

In [11]:
theta_i = 55./360*2*np.pi

calculate all the cases in a for-loop.

In [12]:
TBH = np.empty([len(Freq), 2]) # horizontal channel for 10 and 20 degC cases
TBV = np.empty([len(Freq), 2])
for i in range(len(Freq)):
    TBH10, TBV10 = TBright_calc(theta_i, N_10[i], 10+273.15)
    TBH20, TBV20 = TBright_calc(theta_i, N_20[i], 20+273.15)
    TBH[i, :] = [TBH10, TBH20]
    TBV[i, :] = [TBV10, TBV20]

Show the result.

In [13]:
print('Brightness temperature: \n==============================\ni) Horizontal channels\n\t10degC\t\t20degC')
print(TBH)
print('ii) Vertical channels\n\t10degC\t\t20degC')
print(TBV)
print('==============================')

Brightness temperature: 
i) Horizontal channels
	10degC		20degC
[[  65.29532065   67.7465206 ]
 [  67.77121174   69.62032484]
 [  73.8818992    74.1318003 ]
 [  77.90979187   77.25596997]
 [  87.5757405    85.1777878 ]
 [ 118.80285239  112.98584178]]
ii) Vertical channels
	10degC		20degC
[[ 155.74584188  161.52381861]
 [ 160.07532803  164.81423215]
 [ 170.31082826  172.47891809]
 [ 176.72608374  177.59438292]
 [ 191.0977424   189.89451221]
 [ 228.61742715  226.08061735]]


** Sensitivity, find the best channel **

Here I define the sensitivity as:

\begin{equation*}
    \frac{T_B(20^\circ C) - T_B(10^\circ C)}{10^\circ C}
\end{equation*}

And we believe sensible channel is good.

In [14]:
Sen_TBH = (TBH[:, 1] - TBH[:, 0])/10
Sen_TBV = (TBV[:, 1] - TBV[:, 0])/10

compare the results.

In [15]:
print('Frequency: \n{}'.format(Freq))
print('Sensitivity for horizontal channel: \n{}'.format(Sen_TBH))
print('Sensitivity for vertical channel: \n{}'.format(Sen_TBV))

Frequency: 
[  6.93  10.65  18.7   23.8   36.5   89.  ]
Sensitivity for horizontal channel: 
[ 0.24512     0.18491131  0.02499011 -0.06538219 -0.23979527 -0.58170106]
Sensitivity for vertical channel: 
[ 0.57779767  0.47389041  0.21680898  0.08682992 -0.12032302 -0.25368098]


Channels with frequency higher than 23.8 GHz is not recomended, so we will use **6.93 GHz vertical** channel.

# Problem 7.4

We can fill the blank with the following equations:

\begin{equation*}
    \beta_e = \rho k_e = \sigma_eN
\end{equation*}

\begin{equation*}
    \widetilde{\omega} = \frac{\beta_s}{\beta_e}
\end{equation*}

\begin{equation*}
    \sigma_e = k_em
\end{equation*}

\begin{equation*}
    Qe = \frac{\sigma_e}{A}
\end{equation*}

And here I just type the results:

In [16]:
a = np.array([3.89e2, 6.6e21, 2.8e-19 , 1e-4, 0  , 7.3e-26 , 4.8e-4 , 2.83e-23, 0.187  , 0      ]).T
b = np.array([1e3   , 1e10  , 7.07e-14, 0.2 , 0.1, 1.41e-17, 1.41e-7, 1.41e-14, 1.41e-4, 1.41e-5]).T
c = np.array([0.45  , 80    , 3.14e-6 , 0.6 , 0.4, 4.19e-6 , 3.35e-4, 1.89e-6 , 1.51e-4, 6.03e-5]).T
d = np.array([150   , 1e9   , 3.14e-10, 2   , 0.9, 4.19e-12, 4.19e-3, 6.28e-10, 0.628  , 0.565  ]).T

In [17]:
import csv
import pandas
from itertools import izip

In [18]:
with open('eosc582_a2_p74.csv', 'wb') as f:
    writer = csv.writer(f)
    writer.writerows(izip(a, b, c, d))

In [19]:
pd_table = pandas.read_csv('eosc582_a2_p74.csv', names=['(a)', '(b)', '(c)', '(d)'])
pd_table

(a)           (b)        (c)           (d)
0  3.890000e+02  1.000000e+03   0.450000  1.500000e+02
1  6.600000e+21  1.000000e+10  80.000000  1.000000e+09
2  2.800000e-19  7.070000e-14   0.000003  3.140000e-10
3  1.000000e-04  2.000000e-01   0.600000  2.000000e+00
4  0.000000e+00  1.000000e-01   0.400000  9.000000e-01
5  7.300000e-26  1.410000e-17   0.000004  4.190000e-12
6  4.800000e-04  1.410000e-07   0.000335  4.190000e-03
7  2.830000e-23  1.410000e-14   0.000002  6.280000e-10
8  1.870000e-01  1.410000e-04   0.000151  6.280000e-01
9  0.000000e+00  1.410000e-05   0.000060  5.650000e-01